trajopt_sco/include/trajopt_sco/solver_interface.hpp

```diff
@@ -148,5 +149,53 @@ std::ostream& operator<<(std::ostream&, const Cnt&);
 std::ostream& operator<<(std::ostream&, const AffExpr&);
 std::ostream& operator<<(std::ostream&, const QuadExpr&);
 
-ModelPtr createModel();
+class ConvexSolver
```

Comment:
Do we actually need a whole class for this given that it seems to just wrap an enum?

Reply:
This was a solution I came up to have a place to easily manage conversions from string and to string, for logging and loading of the enum from json (strings are more readable than ints!).
I am open for suggestions in case we can make it more elegant!

Reply:
I plan to look at something similar to what is shown [here](http://www.dorodnic.com/blog/2014/03/29/abstract-factory/)  but this can be done in a separate PR.